### DISCUSSION TITLE
Questions about turtle upgrades JSON files

### DISCUSSION THREAD

**Discussion by vico93** [cc-tweaked/CC-Tweaked#1495] [Q&A] [answered]
[From the datapack-example](https://github.com/cc-tweaked/datapack-example/blob/dev/data/minecraft/computercraft/turtle_upgrades/netherite_sword.json): What does exactly `damageMultiplier` and `breakable` meant? Does these arguments need to be present only on jsons to add swords or all kinds of tools (pickaxes, shovels, axes)?

**Comment by SquidDev** [comment, marked as answer]
The [Java documentation](https://tweaked.cc/mc-1.19.x/javadoc/dan200/computercraft/api/turtle/TurtleUpgradeDataProvider.ToolBuilder.html) probably gives the best explanation here:

 - `damageMultiplier`: The amount of damage a swing of this tool will do. This is multiplied by the attack damage attribute to get the final damage.
 - `breakable`: Provide a list of breakable blocks. If not given, the tool can break all blocks. If given, only blocks in this tag, those in the "always breakable" tag and "insta-mine" ones can be broken.

`damageMultiplier` is optional (defaults to 3), `breakable` is required most of the time - take a look at the built-in diamond tools.

**Reply by vico93** [reply]
I will look on it (the JSONs from default upgrades are in the jar, right?)! Thanks!
